### ISSUE TITLE
Generate function freezes

### ISSUE THREAD

**Issue by ivan-kleshnin** [ai/nanoid#122]
```js
import generate from "nanoid/generate"

generate("abc", 4)   // "baac"
generate("abc", "4") // freezes @_@
```

Obviously a wrong usage, but it's a possible case and freezing the browser is... bad.

This happens because of https://github.com/ai/nanoid/blob/master/format.js#L34 `while (true)` loop.
I believe you should apply `Number` to the `size` argument somewhere.

**Comment by ai** [comment]
Possible solutions:

1. Add warning for development environment as we do here https://github.com/ai/nanoid/blob/master/index.browser.js#L1-L8
2. If it was the real case I can think to add some `+size`

**Comment by ivan-kleshnin** [comment]
What do you mean by "real case"? I surely get this bug by practice, not by reading the source code :)

I would definitely go with 2. as warning won't prevent freezing and you won't be able to even see this warning most probably.

**Comment by ai** [comment]
It will be an error, not a warning. The main problem is different behavior in production and development. But freeze is very close to freeze behavior.

**Comment by ivan-kleshnin** [comment]
> But freeze is very close to freeze behavior.

Not sure what that means. I understand that you don't want to add attribute checking but this seems to be the only case that can cause such behavior.

**Comment by ai** [comment]
I do not want to add `parseInt` code, because it will show down generation (and also increase size, but it is not so critical).

The second reason is more complicated. One extra check will not make us type safe. Only TypeScript can protect from errors like this. 

I will try to add `size+` and check benchmark. Maybe V8 have some smart optimizations for `int+` notion for asm.js?

**Comment by ivan-kleshnin** [comment]
> The second reason is more complicated. One extra check will not make us type safe. Only TypeScript can protect from errors like this.

I know. But errors causing dead freezing should be treated specially IMO. 

> Maybe V8 have some smart optimizations for int+ notion for asm.js?

That's possible. I dunno, the time of this op is not multiplied by a loop or something. It shouldn't affect the behavior in any visible way.

**Comment by ai** [comment]
Fixed 8317d9c

**Comment by ai** [comment]
Is it OK?

**Comment by ivan-kleshnin** [comment]
Yep, LGTM 👍

**Comment by ai** [comment]
Released in 2.0.3
